typescript/sdk/src/hook/EvmHookReader.ts

```diff
@@ -345,4 +359,16 @@ export class EvmHookReader implements HookReader {
       type: HookType.PAUSABLE,
     };
   }
+
+  /**
+   * Conditionally sets the log level for a smart provider.
+   *
+   * @param level - The log level to set, e.g. 'debug', 'info', 'warn', 'error'.
+   */
+  protected setSmartProviderLogLevel(level: string) {
```

Comment:
this also exists in EvmWarpReader. Will DRY it out in a separate PR

Reply:
Any reason we can't do that in this PR?

Reply:
mostly because I haven't figured out a good place to put this yet. 

I need to change the CoreReader to do the same, so i'll need to add this fn in it too.